pkg/gpu/nvidia/manager.go

```diff
@@ -339,5 +448,18 @@ func (ngm *nvidiaGPUManager) Stop() error {
 	}
 	ngm.stop <- true
 	<-ngm.stop
+	if ngm.healthCheckRunning {
+		ngm.healthCheckStop <- true
+		<-ngm.healthCheckStop
```

Comment:
What do we need this for?

Reply:
without this the unit test will fail, because if the health monitoring is not enabled, `nvidiaGPUManager.Stop()` will be blocked indefinitely on `<-ngm.healthCheckStop`. So we need a way for `nvidiaGPUManager.Stop()` to know if it needs to wait for `healthCheckStop` channel to close